pennylane/ops/op_math/exp.py

```diff
@@ -168,14 +168,17 @@ def _queue_category(self):
     def matrix(self, wire_order=None):
 
         coeff_interface = math.get_interface(self.coeff)
-        if coeff_interface == "autograd":
+        if coeff_interface == "autograd" and math.requires_grad(self.coeff):
```

Comment:
Why does it only concern differentiable coefficient?

Reply:
As mentioned in the comment, `math.expm` is not differentiable with autograd.  Therefore if we need the matrix to be differentiable with autograd, we have to construct the matrix via another route.